CMakeLists.txt

```diff
@@ -274,8 +274,28 @@ if(CMAKE_C_COMPILER_ID MATCHES "GNU|AppleClang|Clang")
         "The compiler ${CMAKE_C_COMPILER_ID} does not support -fvisibility=hidden"
     )
   endif(NOT CC_SUPPORTS_VISIBILITY_HIDDEN)
+  # security options go here - these are not supported by all compilers
+  # first and foremost, do we have LTO?
+  check_c_compiler_flag(-flto CC_SUPPORTS_LTO)
+  # if we are building on a GLIBC environment, FORTIFY is available
+  check_c_compiler_flag(-D_FORTIFY_SOURCE CC_SUPPORTS_FORTIFY)
```

Comment:
Can this ever fail? Since you're just setting the `_FORTIFY_SOURCE` preprocessor symbol, it should not be possible for this to fail since the flag is always accepted for C compilers.

You probably need to construct a small program that should generate warning, compile it, and set this flag if the program emits a warning. Something like this:

```suggestion
include(CheckCSourceCompiles)

set(CMAKE_REQUIRED_FLAGS -Werror)
set(CMAKE_REQUIRED_DEFINITIONS -D_FORTIFY_SOURCE)
# If this build succeeds, _FORTIFY_SOURCE is not available
check_c_source_compiles("
#include <string.h>

int main() {
  char buffer[8];

  strcpy(buffer, \"hello world\");
}
" FORTIFY_NOT_AVAILABLE)
```